website/addons/osfstorage/views.py

```diff
@@ -92,8 +93,10 @@ def osfstorage_copy_hook(source, destination, name=None, **kwargs):
 
 @decorators.waterbutler_opt_hook
 def osfstorage_move_hook(source, destination, name=None, **kwargs):
-    return source.move_under(destination, name=name).serialize(), httplib.OK
-
+    try:
+        return source.move_under(destination, name=name).serialize(), httplib.OK
+    except exceptions.FileNodeorChildCheckedOutError:
+        raise HTTPError(httplib.METHOD_NOT_ALLOWED)
```

Comment:
@hmoco Did you add this test?

Also, please add a user-friendly message by passing the `data` dict (see other usages of `HTTPError`), e.g. "File is already checked out".